service/build.go

```diff
@@ -66,7 +67,9 @@ func buildFileStore(cfg *config.Config) (filestore.FileStore, error) {
 	customTransport := &http.Transport{
 	    TLSClientConfig: &tls.Config{InsecureSkipVerify: cfg.S3DisableSSLVerify},
         }
-	customHTTPClient := &http.Client{Transport: customTransport}
+	customHTTPClient := &http.Client{
+                Transport:        customTransport,
+		Timeout:         24 * time.Hour }
```

Comment:
I still need to make this refer to a global config variable (that is also used by the server).